net-misc/sslh/sslh-1.18-r1.ebuild

```diff
@@ -0,0 +1,60 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+MY_P="${PN}-v${PV}"
+inherit toolchain-funcs systemd
+
+DESCRIPTION="Port multiplexer - accept both HTTPS and SSH connections on the same port"
+HOMEPAGE="http://www.rutschle.net/tech/sslh.shtml"
+SRC_URI="http://www.rutschle.net/tech/${PN}/${MY_P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~m68k ~mips ~s390 ~sh ~x86"
+IUSE="caps tcpd systemd"
+
+RDEPEND="caps? ( sys-libs/libcap )
+	tcpd? ( sys-apps/tcp-wrappers )
+	systemd? ( sys-apps/systemd )
+	dev-libs/libconfig"
+DEPEND="${RDEPEND}
+	dev-lang/perl"
+
+RESTRICT="test"
+
+S=${WORKDIR}/${MY_P}
+
+PATCHES=(
+	"${FILESDIR}/${PN}-1.17-version-deps.patch"
+	"${FILESDIR}/${PN}-1.18-version-deps.patch"
+	"${FILESDIR}/${PN}-1.18-systemd-generator.patch"
+)
+
+src_compile() {
+	emake \
+		CC="$(tc-getCC)" \
+		USELIBCAP=$(usev caps) \
+		USELIBWRAP=$(usev tcpd) \
+		USESYSTEMD=$(usev systemd)
+}
+
+src_install() {
+	dosbin sslh-{fork,select}
+	dosym sslh-fork /usr/sbin/sslh
+	doman sslh.8.gz
+	dodoc ChangeLog README.md
+
+	newinitd "${FILESDIR}"/sslh.init.d-2 sslh
+	newconfd "${FILESDIR}"/sslh.conf.d-2 sslh
+
+	# Gentoo puts the binaries in /usr/sbin, but upstream puts them in /usr/bin
+	sed -i -e 's~/usr/bin/~/usr/sbin/~g' scripts/systemd.sslh.service || die
+	systemd_newunit scripts/systemd.sslh.service sslh.service
```

Comment:
Is the unit useful without the generator? If not, it's probably better to make the above two lines conditional too.